### PR TITLE
Disallow new `abstract` type members

### PR DESCRIPTION
**Pull request by dougbu**
- #146
- also add tests of field removals

nits:
- whitespace cleanup in `ApiListingGenerator` and `ApiListingComparerTests`
- remove unused `additionalFilters` parameter from `ApiListingComparerTests.CreateApiListingDocument()`

### CODE REVIEW

File: test/ApiCheck.Test/ApiListingComparerTests.cs [hunk 1]
```diff
@@ -80,6 +79,68 @@ public void Compare_Detects_TypeGenericAritybreakingChanges_as_removal()
...
+        [InlineData("public struct ComparisonScenarios.StructToRemoveFieldsFrom")]
+        public void Compare_DetectsAllFieldRemovals(string typeToCheck)
+        {
+            // Arrange
+            var v1ApiListing = CreateApiListingDocument(V1Assembly);
+            var v2ApiListing = CreateApiListingDocument(V2Assembly);
+            var comparer = new ApiListingComparer(v1ApiListing, v2ApiListing);
+
+            // Oops. The NewInternalProperty addition is a breaking change; makes it impossible to subclass type in
+            // another assembly.
```
Comment (dougbu) [inline test/ApiCheck.Test/ApiListingComparerTests.cs]: @Eilon these "oops" comments are about corner cases. Worth filing bugs?

Comment (Eilon) [inline test/ApiCheck.Test/ApiListingComparerTests.cs]: Sorry I don't understand the comment in the code. What is "oops" about it? And what is the issue?

Comment (dougbu) [inline test/ApiCheck.Test/ApiListingComparerTests.cs]: API check doesn't complain when an `internal abstract` member is added though that is a significant breaking change -- prevents creation of a subclass outside the defining assembly. As I said, it's a corner case but "oops", something is slipping though&hellip;

File: test/ApiCheck.Test/ApiListingComparerTests.cs [hunk 2]
```diff
@@ -80,6 +79,68 @@ public void Compare_Detects_TypeGenericAritybreakingChanges_as_removal()
...
+                new BreakingChange(
+                    typeToCheck,
+                    "public const System.Int32 ConstToChangeValue = 1",
+                    ChangeKind.Removal),
+                // Removing a const doesn't cause a binary incompatibilty but often causes problems.
+                new BreakingChange(
+                    typeToCheck,
+                    "public const System.Int32 ConstToMakeField = 2",
+                    ChangeKind.Removal),
+                // Oops. Making a field writable is not technically a breaking change.
```
Comment (dougbu) [inline test/ApiCheck.Test/ApiListingComparerTests.cs]: And the oops here is that API Check is disallowing a non-breaking change to a field. This one is also a corner case and it's also easy to work around (with entries in the breaking change files).

**Comment by dougbu** [comment]
3596a1f
